src/app/cdk/panel/panel/panel.component.ts

```diff
@@ -117,7 +120,7 @@ export class PanelComponent implements OnInit {
     return obj && (obj.value || obj.putCode)
   }
 
-  openModal() {
+  openModal(isCopy: boolean) {
```

Comment:
```suggestion
  openModal(options:{ createACopy: boolean = false} ) {
```
This makes this option easier to understand when reading the code from the template. Also might only be required when the user is making a copy by defaulting to false.

Reply:
makes sense. I will change it